docs/source/docker/advanced/datasource_predictions.rst

```diff
@@ -0,0 +1,347 @@
+of the number of frames in the video) separated by hyphens:
+
+.. code-block:: bash
+
+    # filename of the predictions of the Xth frame of video FILENAME.EXT
+    # with 200 frames (padding: len(str(200)) = 3)
+    .lightly/predictions/${TASK_NAME}/${FILENAME}-${X:03d}-${EXT}.json
+
+    # example: my_video.mp4, frame 99/200
+    .lightly/predictions/my_classification_task/my_video-099-mp4.json
+
+    # example: my_subdir/my_video.mp4, frame 99/200
+    .lightly/predictions/my_classification_task/my_subdir/my_video-099-mp4.json
+
+
+Prediction Format
+-----------------
+Predictions for an image must have a `file_name` and `predictions`.
+Here, `file_name` serves as a unique identifier to retrieve the image for which
+the predictions are made and predictions is a list of `Prediction Singletons` for the corresponding task.
+
+Example classification:
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/classification_weather/my_image.json
+
+    {
+        "file_name": "my_image.png"
+        "predictions": [ // classes: [sunny, clouded, rainy]
+            {
+                "category_id": 0,
+                "probabilities": [0.8, 0.1, 0.1]
+            }
+        ]
+    }
+
+Example object detection:
+
+.. code-block:: javascript
+    :caption: .lightly/predictions/object_detection/my_image.json
+
+    {
+        "file_name": "my_image.png",
+        "predictions": [ // classes: [person, car]
+            {
+                "category_id": 0,
+                "bbox": [...],
+                "score": 0.8
+            },
+            {
+                "category_id": 1,
+                "bbox": [...],
+                "score": 0.9
+            },
+            {
+                "category_id": 0,
+                "bbox": [...],
+                "score": 0.5
+            }
+        ]
+    }
+
+Note: The filename should always be the full path from the root directory.
+
+
+Prediction Singletons
+---------------------
+The prediction singletons closely follow the `COCO results <https://cocodataset.org/#format-results>`_ format while dropping
+the `image_id`. Note the the `category_id` must be the same as the one defined
+in the schema and that the probabilities (if provided) must follow the order of the category ids.
+
+**Classification:**
+
+For classification, please use the following format:
+
+.. code-block:: javascript
+
+    [{
+        "category_id"       : int,
+        "probabilities"     : [p0, p1, ..., pN],
+    }]
+
+**Object Detection:**
+
+For detection with bounding boxes, please use the following format:
+
+.. code-block:: javascript
+
+    [{
+        "category_id"       : int,
+        "bbox"              : [x, y, width, height],
+        "score"             : float,
+        "probabilities"     : [p0, p1, ..., pN],         // optional
+    }]
+
+The bounding box format follows the `COCO results <https://cocodataset.org/#format-results>`_ documentation.
```

Comment:
We should mention here that segmentation and keypoints are not supported yet. Either add a coming soon or an info box.